test/cmpd_dset.c

```diff
@@ -2210,6 +2210,46 @@ test_ooo_order(char *filename, hid_t fapl_id)
     return 1;
 } /* test_ooo_order */
 
+static unsigned
+test_compound_error_member()
+{
+    hid_t tenum = -1;
+    hid_t tcmp  = -1;
+    TESTING("that error on compound member type fail early");
+
+    if ((tenum = H5Tenum_create(H5Tcopy(H5T_NATIVE_INT))) < 0)
+        TEST_ERROR;
+
+    if ((tcmp = H5Tcreate(H5T_COMPOUND, 2 * H5Tget_size(H5T_NATIVE_LONG))) < 0)
+        TEST_ERROR;
+
+    if ((H5Tinsert(tcmp, "long", 0, H5T_NATIVE_LONG)) < 0)
+        TEST_ERROR;
+
+    if ((H5Tinsert(tcmp, "enum", H5Tget_size(H5T_NATIVE_LONG), tenum)) < 0)
```

Comment:
Is this call supposed to fail? If so, why does the test pass if it doesn't (and fail if it does)?